src/script/components/MessagesList/MessageList.tsx

```diff
@@ -245,78 +255,167 @@ export const MessagesList: FC<MessagesListParams> = ({
+                  isLastDeliveredMessage={isLastDeliveredMessage}
+                  isHighlighted={isHighlighted}
+                  scrollTo={scrollToElement}
+                  isSelfTemporaryGuest={selfUser.isTemporaryGuest()}
+                  messageRepository={messageRepository}
+                  onClickAvatar={showUserDetails}
+                  onClickCancelRequest={cancelConnectionRequest}
+                  onClickImage={showImageDetails}
+                  onClickInvitePeople={() => invitePeople(conversation)}
+                  onClickReactionDetails={message => showMessageReactions(message, true)}
+                  onClickMessage={onClickMessage}
+                  onClickParticipants={showParticipants}
+                  onClickDetails={message => showMessageDetails(message)}
+                  onClickResetSession={resetSession}
+                  onClickTimestamp={async function (messageId: string) {
+                    setHighlightedMessage(messageId);
+                    setTimeout(() => setHighlightedMessage(undefined), 5000);
+                    const messageIsLoaded = conversation.getMessage(messageId);
+
+                    if (!messageIsLoaded) {
+                      const messageEntity = await messageRepository.getMessageInConversationById(
+                        conversation,
+                        messageId,
+                      );
+                      conversation.removeMessages();
+                      conversationRepository.getMessagesWithOffset(conversation, messageEntity);
+                    }
+                  }}
+                  selfId={selfUser.qualifiedId}
+                  shouldShowInvitePeople={shouldShowInvitePeople}
+                  isFocused={isFocused}
+                  handleFocus={setFocusedId}
+                  handleArrowKeyDown={handleKeyDown}
+                  isMsgElementsFocusable={isMsgElementsFocusable}
+                  setMsgElementsFocusable={setMsgElementsFocusable}
+                />
+              );
+            });
+          })}
+        </div>
+      </FadingScrollbar>
+      <JumpToLastMessageButton onGoToLastMessage={jumpToLastMessage} conversation={conversation} />
+    </>
+  );
+};
+
+export interface JumpToLastMessageButtonProps extends HTMLProps<HTMLElement> {
+  onGoToLastMessage: () => void;
+  conversation: Conversation;
+}
+
+export const JumpToLastMessageButton: FC<JumpToLastMessageButtonProps> = ({
+  onGoToLastMessage,
+  conversation,
+}: JumpToLastMessageButtonProps) => {
+  // To be changed when design chooses a breakpoint, the conditional can be integrated to the ui-kit directly
+  const mdBreakpoint = useMatchMedia('max-width: 768px');
+
+  const [isLastMessageVisible, setIsLastMessageVisible] = useState(conversation.isLastMessageVisible());
+
+  useEffect(() => {
+    const subscription = conversation.isLastMessageVisible.subscribe(
+      debounce(value => {
```

Comment:
debouncing is needed for messages, sent by `self`: at first the message from self gets mounted as not yet delivered, then it gets unmounted and mounted as received from the server. this makes the jump to the last message button blink. there is a complicated way to check if a message that is unmounted (its `onVisibilityLost` was called) has not yet been delivered, but we should somehow distinguish whether it was unmounted and not hidden by scrolling. I think debouncing is better, while it does add a bit of noticeable delay even with 200 ms